apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -27,7 +28,7 @@ defmodule Aecore.Chain.Worker do
     txs_index = calculate_block_acc_txs_info(Block.genesis_block())
     initial_state = {genesis_block_map, latest_block_chain_state, txs_index}
 
-    {:ok, initial_state}
+    {:ok, initial_state, 2000}
```

Comment:
Could you elaborate why do we wait 2sec. after start to load the chain from disk? Is this because `Aecore.Persistence.Worker` may not be started?

If so we should find another solution of the problem. I'm not aware of any clean way in Elixir to make `Chain.Worker` depend on `Persistance.Worker`, so I would even suggest merging `Persistance.Worker` with `Chain.Worker`.

Reply:
Yes, the reason is `Aecore.Persistence.Worker` may not be started yet. I will search for better solution.

Reply:
@meivantodorov any progress on this?